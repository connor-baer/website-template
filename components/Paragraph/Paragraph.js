import styled from '@emotion/styled';
import { css } from '@emotion/core';

const baseStyles = () => css`
  line-height: 1.75rem;
  margin: 1rem 0;
`;

const Paragraph = styled('p')(baseStyles);

export default Paragraph;
