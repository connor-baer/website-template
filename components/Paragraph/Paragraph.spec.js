import React from 'react';

import Paragraph from './Paragraph';

describe('Paragraph', () => {
  function renderParagraph(fn, props = {}) {
    return fn(<Paragraph {...props} />);
  }

  it('should render with default styles', () => {
    const children = 'Bacon ipsum dolor amet swine shank biltong ham bacon.';
    const { container } = renderParagraph(render, { children });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should meet basic accessibility guidelines', async () => {
    const children = 'Bacon ipsum dolor amet swine shank biltong ham bacon.';
    const html = renderParagraph(renderToHtml, { children });
    const result = await axe(html);
    expect(result).toHaveNoViolations();
  });
});
