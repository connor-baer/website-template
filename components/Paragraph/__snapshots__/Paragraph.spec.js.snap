// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph should render with default styles 1`] = `
.circuit-0 {
  line-height: 1.75rem;
  margin: 1rem 0;
}

<p
  class="circuit-0 circuit-1"
>
  Bacon ipsum dolor amet swine shank biltong ham bacon.
</p>
`;
